packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1263,7 +1276,7 @@ export class ContainerRuntime
 			this.remoteMessageProcessor.clearPartialMessagesFor(clientId);
 		});
 
-		this.summaryCollection = new SummaryCollection(this.deltaManager, this.logger);
+		this.summaryCollection = new SummaryCollection(this._deltaManager, this.logger);
```

Comment:
It would be great to use proxy object (through getter) everywhere except where we actually need real thing. That way it will be clear where we have dependencies in summarizer on real thing.
Looking at this code (and usage of same _deltaManager on line 1152) I do not know if this is intentional, or this code does not care

Reply:
We could do that and only pass in real delta manager where needed. This means even within container runtime, the usages will be split between the proxy and the real delta manager. I think this makes it confusing as to which one is supposed to be used where when someone wants to add a new dependency on delta manager. So, I thought we could keep it simple - withing container runtime, use the real delta manager and every where else, use the proxy.

Reply:
I don't see a clean solution either way. There is no easy way to say `don't use the getter if you want the real delta manager` except for adding comments which are easy to miss.